### ISSUE TITLE
Request: remove option "type" by default

### ISSUE THREAD

**Issue by dionjwa** [c4spar/deno-cliffy#127]
Here's a sample output of a command I'm building:

<img width="540" alt="image" src="https://user-images.githubusercontent.com/260221/103159305-bac63200-477c-11eb-9fad-cbe8ef7929f7.png">

Problem: I find the "type" of options printed out:
- useless information: almost always the option is a string
- not helpful: users don't need to know what a "string" is, or that you need to think about it
- visual clutter: I have a list of options for users to visually parse, the "type" is nested in there, but is actually something else entirely, even though it's a different colour. 

Most CLI tools don't bother listing the type, and `cliffy` has great API support for parsing and validating options, so e.g. if a user needed to enter a valid email or number, and it's not valid, the command throws an error explaining how to fix it. This to me is a much better UX: minimize the information a user needs to use the tool, and provide helpful errors when the input is not valid. 

I would like, in order of preference:
1. The "type" information hidden by default
2. An option to show the type information
3. A way to customize that help info

This is a awesome tool, it's hard to get a generic powerful CLI creation tool, and this one is really good, so excellent work  👍 

If this isn't easy, I'll see about looking into the code myself, if that's your preference, some code hints of where to look would be really helpful.

**Comment by c4spar** [comment]
Hi @dionjwa, thanks for your great feedback! 🙂 

I was already thinking about to remove the types by default and add some options to configure colors and hints. But i didn't have decided yet how to implement this. If you want to implement that you are welcome to do so 👍 

The think is, the help option, the help command and the getHelp method should play well together.

Currently the Command class has a getHelp method which uses the HelpGenerator to generate and return the help output. The help option and help command uses both the getHelp method to print the help output.

At the moment you can override the output of the help option with:
```typescript
new Command()
  .helpOption("-h, --help", "Show this help.", () => {
    console.log("my custom help");
  })
```
But this will only override the output for the help option and not for the getHelp method and the help command.

One option could be to add a `.help()` method to the `Command` class, similar to the `.version()` method, which allows you to override the output of the `.getHelp()` method. The implementaion could look like this:
```typescript
protected _help: (this: Command<O, A>) => string = () => HelpGenerator.generate(this);

public help(help: string | ((this: Command<O, A>) => string) | { colors: boolean; types: boolean }): this {
  switch(typeof help) {
    case "string":
      this.cmd._help = () => help;
      break;
    case "function":
      this.cmd._help = help;
      break;
    default:
      this.cmd._help = () => HelpGenerator.generate(this, help);
  }
  return this;
}

public getHelp(): string {
  return this._help();
}
```
If a string is passed the string will be use, if a funtion is passed the return value will be used and if an options object is passed the options are passed to the default help generator. The `.getHelp()` method than calls simply the `._help()` method and returns the result.

But there is already a `.help()` method which prints the help text to stdout. So i think i would rename it to something like `.showHelp()` or `.printHelp()`.

What do you think?

**Comment by c4spar** [comment]
But i think we have to set the default value of the `_help` property inside the `.registerDefaults()` method to be sure its only set on the main command.
The `.getHelp()` method than could look like this:
```typescript
public getHelp(): string {
  return this._help?.() ?? this._parent?._getHelp();
}
```

**Comment by c4spar** [comment]
hi @dionjwa,
the types are now hidden by default in the help output and you can also configure and override the help output.
It's available in v0.17.0.
